R/filters.R

```diff
@@ -0,0 +1,23 @@
+# Possible filters
+
+#' Create a filter for pgRNAs which have a raw count of 0 for any sample/time point
+#' @description This function flags and reports which and how many pgRNAs have a raw count of 0 for any sample/time point
+#' @param gimap_dataset The special gimap_dataset from the `setup_data` function which contains the transformed data
+#' @importFrom magrittr %>%
+#' @importFrom dplyr mutate
+#' @return a named list with the filter `filter` specifying which pgRNA have a count zero for at least one sample/time point and a report df `reportdf` for the number and percent of pgRNA which have a count zero for at least one sample/time point 
+#' @examples \dontrun{
+#' 
+#' }
+#'
+
+qc_filter_zerocounts <- function(gimap_dataset){
+  
+  counts_filter <- unlist(lapply(1:nrow(gimap_dataset$raw_counts), function(x) 0 %in% gimap_dataset$raw_counts[x,]))
```

Comment:
Can you explain the goal of this line?

Reply:
The goal of this line is to create a vector that reports for each pgRNA construct whether any one of the samples has a count of 0. I used `lapply` to loop through each row/pgRNA construct and for each row check for a zero and then `unlist` to make sure it was a simple vector, not nested

Reply:
I figured out a tidyverse way to do it!  Do you prefer the tidyverse way below or a different way all together?
```
counts_filter <- data.frame(gimap_dataset$raw_counts) %>% map(~.x %in% c(0)) %>% reduce(`|`)
```

Reply:
I like this tidyverse way! I think its a bit more readable! Can you transfer this in?